libr/core/linux_heap_glibc.c

```diff
@@ -303,19 +337,43 @@ static void GH(print_arena_stats)(RCore *core, GHT m_arena, MallocState *main_ar
 }
 
 static bool GH(r_resolve_main_arena)(RCore *core, GHT *m_arena) {
+
 	if (!core || !core->dbg || !core->dbg->maps) {
 		return false;
 	}
 
+	if (main_arena_resolved) {
+		return true;
+	}
+
+	const char *main_arena_str = "main_arena";
 	GHT brk_start = GHT_MAX, brk_end = GHT_MAX;
 	GHT libc_addr_sta = GHT_MAX, libc_addr_end = 0;
 	GHT addr_srch = GHT_MAX, heap_sz = GHT_MAX;
+	const char *libc_path = NULL;
+	GHT libc_addr = GHT_MAX;
+	GHT main_arena_sym = 0;
 
 	if (r_config_get_i (core->config, "cfg.debug")) {
 		RListIter *iter;
 		RDebugMap *map;
 		r_debug_map_sync (core->dbg);
 		r_list_foreach (core->dbg->maps, iter, map) {
+			if (strstr (map->name, "/libc-") && map->perm == 4 && !main_arena_sym) {
+				libc_path = map->name;
+				libc_addr = map->addr;
+				if (!libc_path) {
+					break;
+				}
+				char *path = strdup(libc_path);
+				if (r_file_exists (path)) {
+					GHT vaddr = GH (get_va_symbol) (path, main_arena_str);
+					if (libc_addr != GHT_MAX && vaddr != GHT_MAX) {
+						main_arena_sym = libc_addr + vaddr;
+						free (path);
```

Comment:
uhm this free seems to be placed in the wrong line. it will be leaking its value in some situations